### PR TITLE
Handle locks.

### PR DESCRIPTION
**Pull request by alexpott**

### CODE REVIEW

File: core/tests/Drupal/Nightwatch/globals.js [hunk 1]
```diff
@@ -30,6 +30,14 @@ module.exports = {
     if (phpWebServer) {
       phpWebServer.kill();
     }
+    try {
+      execSync(commandAsWebserver(`php ./scripts/test-site.php release-lock --all`));
+    }
+    catch(error) {
+      this.assert.fail(error);
```
Comment (alexpott) [inline core/tests/Drupal/Nightwatch/globals.js]: This is probably not correct for this context.

File: core/lib/Drupal/Core/Test/TestDatabase.php [hunk 2]
```diff
@@ -107,31 +111,48 @@ public function getDatabasePrefix() {
   /**
    * Generates a unique lock ID for the test method.
    *
+   * @param bool $create_lock
+   *   (optional) Whether or not to create a lock file. Defaults to FALSE.
```
Comment (ryanaslett) [inline core/lib/Drupal/Core/Test/TestDatabase.php]: whats the use case for *not* having a lock? why not always do it?

Comment (dawehner) [reply core/lib/Drupal/Core/Test/TestDatabase.php]: Ha, I had the same question :)

**Comment by dawehner** [comment]
This PR is out of date.

**Comment by dawehner** [comment]
It feels like it would be better to get the issue done on drupal.org, give how close it is right now.

**Pull request closed without merging**